libraries/eosiolib/contracts/eosio/key_value.hpp

```diff
@@ -713,14 +684,139 @@ class kv_table {
+
+      template <typename KF>
+      kv_unique_index(eosio::name name, KF&& kf) : kv_index{name, kf} {}
+
+      /**
+       * Search for an existing object in a table by the index, using the given key.
+       * @ingroup keyvalue
+       *
+       * @tparam K - The type of the key. This will be auto-deduced by the key param.
+       *
+       * @param key - The key to search for.
+       * @return An iterator to the found object OR the `end` iterator if the given key was not found.
+       */
+      template <typename K>
+      iterator find(K&& key) {
+         auto t_key = table_key(prefix, make_key(std::forward<K>(key)));
+
+         uint32_t itr = internal_use_do_not_use::kv_it_create(tbl->db_name, contract_name.value, prefix.data(), prefix.size());
+         int32_t itr_stat = internal_use_do_not_use::kv_it_lower_bound(itr, t_key.data(), t_key.size());
+
+         auto cmp = internal_use_do_not_use::kv_it_key_compare(itr, t_key.data(), t_key.size());
+
+         if (cmp != 0) {
+            internal_use_do_not_use::kv_it_destroy(itr);
+            return this->end();
+         }
+
+         return {contract_name, itr, static_cast<kv_it_stat>(itr_stat), this};
+      }
+
+      /**
+       * Get the value for an existing object in a table by the index, using the given key.
+       * @ingroup keyvalue
+       *
+       * @tparam K - The type of the key. This will be auto-deduced by the key param.
+       *
+       * @param key - The key to search for.
+       * @return A std::optional of the value corresponding to the key.
+       */
+      template <typename K>
+      std::optional<T> get(K&& key) {
+         uint32_t value_size;
+         std::optional<T> ret_val;
+
+         auto t_key = table_key(prefix, make_key(std::forward<K>(key)));
+
+         auto success = internal_use_do_not_use::kv_get(tbl->db_name, contract_name.value, t_key.data(), t_key.size(), value_size);
+         if (!success) {
+            return ret_val;
+         }
+
+         void* buffer = value_size > detail::max_stack_buffer_size ? malloc(value_size) : alloca(value_size);
+         auto copy_size = internal_use_do_not_use::kv_get_data(tbl->db_name, 0, (char*)buffer, value_size);
+
+         void* deserialize_buffer = buffer;
+         size_t deserialize_size = copy_size;
+
+         if (this->name != tbl->primary_index->name) {
+            uint32_t actual_data_size;
+            auto success = internal_use_do_not_use::kv_get(tbl->db_name, contract_name.value, (char*)buffer, copy_size, actual_data_size);
+            eosio::check(success, "failure getting primary key");
+
+            void* pk_buffer = actual_data_size > detail::max_stack_buffer_size ? malloc(actual_data_size) : alloca(actual_data_size);
+            auto pk_copy_size = internal_use_do_not_use::kv_get_data(tbl->db_name, 0, (char*)pk_buffer, actual_data_size);
+
+            eosio::check(pk_copy_size != copy_size, "failure getting primary index data");
```

Comment:
I'm sure I mentioned this before, but I don't see the comment anywhere.  This check is comparing the size of the primary key to the size of the secondary key, which is not a meaningful comparison.

Reply:
You did mention it, and I thought I removed it. I think I messed up a merge somewhere. Regardless, I'll remove it again.